File: app/src/main/java/com/code_design_camp/client/friday/HeadDisplayClient/preference/ThemeSelectPreference.java
package com.code_design_camp.client.friday.HeadDisplayClient.preference;

import android.content.Context;
import android.util.AttributeSet;

import androidx.fragment.app.FragmentManager;
import androidx.preference.Preference;

import com.code_design_camp.client.friday.HeadDisplayClient.dialog.ThemeDialog;
import com.code_design_camp.client.friday.HeadDisplayClient.ui.SettingsActivity;

public class ThemeSelectPreference extends Preference {
    ThemeDialog d = new ThemeDialog();

    public ThemeSelectPreference(Context c, AttributeSet attrs) {
        super(c, attrs);
    }

    public void showDialog(ThemeDialog.OnSelectedTheme listener) {
        FragmentManager fm = ((SettingsActivity) getContext()).getSupportFragmentManager();
        d.setOnApplyThemeListener(listener);
        d.show(fm, "test");
    }
}

File: appModules/preferences/src/main/java/com/friday/ar/preferences/ui/preference/ThemeSelectPreference.kt
package com.friday.ar.preferences.ui.preference

import android.content.Context
import android.util.AttributeSet

import androidx.preference.Preference

import com.friday.ar.preferences.ui.SettingsActivity
import com.friday.ar.preferences.ui.dialog.ThemeDialog

class ThemeSelectPreference(c: Context, attrs: AttributeSet) : Preference(c, attrs) {
    internal var d = ThemeDialog()

    fun showDialog(listener: ThemeDialog.OnSelectedTheme) {
        val fm = (context as SettingsActivity).supportFragmentManager
        d.setOnApplyThemeListener(listener)
        d.show(fm, "test")
    }
}
